miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,590 @@
+    "device_fault": {"siid": 2, "piid": 2},
+    "device_status": {"siid": 2, "piid": 1},
+    "filter_life_level": {"siid": 10, "piid": 1},
+    "filter_left_time": {"siid": 10, "piid": 2},
+    "brush_left_time": {"siid": 11, "piid": 1},
+    "brush_life_level": {"siid": 11, "piid": 2},
+    "brush_left_time2": {"siid": 12, "piid": 1},
+    "brush_life_level2": {"siid": 12, "piid": 2},
+    "brush_left_time3": {"siid": 15, "piid": 1},
+    "brush_life_level3": {"siid": 15, "piid": 2},
+    "sweep_mode": {"siid": 14, "piid": 1},
+    "cleaning_mode": {"siid": 2, "piid": 4},
+    "sweep_type": {"siid": 2, "piid": 8},
+    "path_mode": {"siid": 13, "piid": 8},
+    "mop_present": {"siid": 8, "piid": 1},
+    "work_station_freq": {"siid": 8, "piid": 2},  # Range: [0, 3, 1]
+    "timing": {"siid": 8, "piid": 6},  # str (example: {"tz":2,"tzs":7200})
+    "clean_area": {"siid": 8, "piid": 7},  # uint32
+    # "uid": {"siid": 8, "piid": 8}, # str
+    "auto_boost": {"siid": 8, "piid": 9},  # bool (auto boost on carpet)
+    "forbid_mode": {"siid": 8, "piid": 10},  # str
+    "water_level": {"siid": 8, "piid": 11},
+    # "siid8_13": {"siid": 8, "piid": 13}, # no-name: (uint32, unit: seconds) (acc: ['read', 'notify'])
+    # "siid8_14": {"siid": 8, "piid": 14}, # no-name: (uint32, unit: none) (acc: ['read', 'notify'])
+    "clean_counts": {"siid": 8, "piid": 18},
+    # "siid8_19": {"siid": 8, "piid": 19}, # no-name: (uint32, unit: seconds) (acc: ['read', 'notify'])
+    "double_clean": {"siid": 8, "piid": 20},
+    "edge_sweep": {"siid": 8, "piid": 21},
+    "led_switch": {"siid": 8, "piid": 22},
+    "lidar_collision": {"siid": 8, "piid": 23},
+    "station_key": {"siid": 8, "piid": 24},
+    "station_led": {"siid": 8, "piid": 25},
+    "current_audio": {"siid": 8, "piid": 26},  # str (example: girl_en)
+    "progress": {"siid": 8, "piid": 28},
+    # "station_type": {"siid": 8, "piid": 29}, # uint32
+    # "voice_conf": {"siid": 8, "piid": 30},
+    # "switch_status": {"siid": 2, "piid": 10},
+    "volume": {"siid": 9, "piid": 1},
+    "mute": {"siid": 9, "piid": 2},
+}
+
+
+class ChargingState(Enum):
+    Unknown = -1
+    Charging = 1
+    Discharging = 2
+    NotChargeable = 4
+
+
+class CleaningMode(Enum):
+    Unknown = -1
+    Silent = 1
+    Basic = 2
+    Strong = 3
+    FullSpeed = 4
+    Sweep = 0
+
+
+class SweepType(Enum):
+    Unknown = -1
+    Sweep = 0
+    Mop = 1
+    MopAndSweep = 2
+
+
+class SwitchStatus(Enum):
+    Unknown = -1
+    Open = 1
+
+
+class PathMode(Enum):
+    Unknown = -1
+    Normal = 0
+    YMopping = 1
+    RepeatMopping = 2
+
+
+class WaterLevel(Enum):
+    Unknown = -1
+    First = 1
+    Second = 2
+    Three = 3
+    Fourth = 4
+    Mop = 0
+
+
+class SweepMode(Enum):
+    Unknown = -1
+    Total = 1
+    Area = 2
+    Curpoint = 3
+    Point = 4
+    Smart = 7
+    AmartArea = 8
+    DepthTotal = 9
+    AlongWall = 10
+    Idle = 0
+
+
+class FaultStatus(Enum):
```

Comment:
It is probably simpler to have this as a dictionary than as an enum (the value being a user-friendly message, see https://github.com/rytilahti/python-miio/blob/master/miio/vacuumcontainers.py#L16).